### ISSUE TITLE
New type of Dell API Key?

### ISSUE THREAD

**Issue by Alwaysin** [InfotelGLPI/manufacturersimports#61]
* Version of the plugin : 
2.1.0

* Version of your GLPI : 
9.3.0

* Steps to reproduce (which actions have you made) :
Adding key of type xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

* Expected result :
Working

* Actual result :
Not working

I had to apply a new key and they gave me this. It is not the same format as the previous key I had. Is it known? Does the plugin need to adapt to this new format?

**Comment by Wolvverine** [comment]
The problem is that they are covered with 3-4 characters api key (which now has 37 characters for example in my case)

**Comment by Wolvverine** [comment]
check your API key:
https://api.dell.com/support/assetinfo/v4/getassetheader/{id}?apikey={apikey}
ID Single service tag 
apikey - your api key

**Comment by Alwaysin** [comment]
@amma35 It looks like I'm still facing the issue with 2.1.1

**Comment by amma35** [comment]
error SQL ?

**Comment by Wolvverine** [comment]
Look on server response , I have now:

This XML file does not appear to have any style information associated
with it. The document tree is shown below.
<Fault xmlns:env="http://schemas.xmlsoap.org/soap/envelope/" xmlns="http://api.dell.com/faults">
<Code>eAPI-401</Code>
<Url>http://developer.dell.com/faults#eAPI-50020</Url>
<Message>Invalid API Key</Message>
<ErrorSubCode>0x00d30003</ErrorSubCode>
<ResourceURI>
https://10.166.24.5:443/support/assetinfo/v4/getassetwarranty/yyyyyyy?apikey=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
</ResourceURI>
</Fault>

W dniu 2018-08-03 o 14:16, Amandine Manceau pisze:

**Comment by amma35** [comment]
Is the DELL API key registered in the configuration? 
The correction made is only the size of the API key input field.

**Comment by Wolvverine** [comment]
Yes. It is link from application.

pt., 3 sie 2018, 16:47 użytkownik Amandine Manceau <notifications@github.com>
napisał:

**Comment by Wolvverine** [comment]
We have processed your “Warranty Status V4 API” request and below is the API key for your use.

 

Warranty V4 API Key
 

Initially it is for use in our sandbox environment. Post successful testing in Sand Box , your key will be promoted to production. Would appreciate if you can keep us posted once you complete Sandbox testing.

In another email you will receive an attached zip file with documentation, sample code, and a copy of license agreement file.

 

Sandbox API End Point URL :  https://sandbox.api.dell.com (The SDK document references production end point(s). You need to change https://api.dell.com to https://sandbox.api.dell.com for testing in Sandbox).

**Comment by amma35** [comment]
Yes it is normal, you have to change the url in the configuration for your tests.

**Comment by yalgva** [comment]
Hello, did you find the solution because I have the same error 
https://sandbox.api.dell.com/9YTCDK2?apikey=7072a692b17748ebbd5f14b18208 
<Fault xmlns:env="http://schemas.xmlsoap.org/soap/envelope/" xmlns="http://api.dell.com/faults">
<Code xmlns="">eAPI-401</Code>
<Message xmlns="">Invalid API Key</Message>
<Reason xmlns="">Authentication failure</Reason>
<Source xmlns="">Gateway</Source>
</Fault>

**Comment by yalgva** [comment]
fixed after 10 minutes

**Comment by Alwaysin** [comment]
@yalgva what is the fix?

**Comment by yalgva** [comment]
Hello I received this message from DEll

Dear Yannick,
 
We are pleased to inform you that the Warranty V4 API key has been promoted for use with our production environment and you can now deploy your application and use the production services.
Once you have deployed and are using the production services, would appreciate if you can notify us so that we can mark your project engagement as completed.


and since yesterday import does not work anymore.

Do I need to modify something in my setup ?

Thank you for your help

**Comment by amma35** [comment]
URL for tests :  https://sandbox.api.dell.com 
API Production : https://api.dell.com/support/assetinfo/v4/getassetwarranty/

**Comment by yalgva** [comment]
Hello

same issue with api production url

**Comment by yalgva** [comment]
What I have to put in "Url du fabricant" in GLPI

https://api.dell.com/support/assetinfo/v4/getassetwarranty/

or anything else ?

**Comment by amma35** [comment]
If no changes have been made in the code, just indicate :
![dell](https://user-images.githubusercontent.com/15138371/46676460-df0bb780-cbe0-11e8-8c4e-d1148b1a50b5.png)
The rest is done in the plugin. 

![dell](https://user-images.githubusercontent.com/15138371/46677055-1d55a680-cbe2-11e8-87ce-a942ac90195b.png)

**Comment by dcarro** [comment]
Hi,
I received a new API key from Dell and when I want to change it, it doesn't work.
I get a new api key longer than previous, always I try to change the key it drop my new key.

After I check the database and I saw the column  supplier_key in glpi_plugin_manufacturersimports_configs was limited to 32 characters. 

I modified like that:
`alter table glpi_plugin_manufacturersimports_configs modify supplier_key VARCHAR(40);`

I hope is useful for someone.

**Comment by scrfc71** [comment]
> I just received my API Key and Key Secret via email. Does Dell **REQUIRE** us to ONLY use **OAUTH2** now? I can't use the below format with my API key and servicetag. I even tried without hypens. I replaced the below X's with my real servicetag and API key:
> 
> https://api.dell.com/support/assetinfo/v4/getassetwarranty/XXXXXXXXXX?apikey=XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
> 
> If I use the above format in my web browser, I get the below error:
> 
> ```
>  <?xml version="1.0" encoding="ISO-8859-1"?>
>  <Fault xmlns="http://api.dell.com/faults" xmlns:env="http://schemas.xmlsoap.org/soap/envelope/">
>  <Code>eAPI-401</Code>
>  <Url>http://developer.dell.com/faults#eAPI-50020</Url>
>  <Message>Invalid API Key</Message>
>  <ErrorSubCode>0x00d30003</ErrorSubCode>
>  <ResourceURI> https://api.dell.com/support/assetinfo/v4/getassetwarranty/XXXXXXXXXX?apikey=XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX</ResourceURI>
>  </Fault>
> ```

Same problem here. DELL API can only be used with OAUTH2. THis script does not work anymore.
Any suggestions?

**Comment by mkanet** [comment]
@scrfc71  yes, Dell now requires OAuth2.  Please look at my post on Reddit which has a fully working solution (as long as you have the API key and respective Key secret).  All you need to do is copy and paste.

[https://www.reddit.com/r/PowerShell/comments/cvu7z9/all_getdellwarrantyps1_users_should_read_this/](url)

**Comment by pablitobin** [comment]
It is a powershell script. I think that the issue is that we are using dell.class.php on GLPI.
